### DISCUSSION TITLE
Is the Lanczos Filter a good decision?

### DISCUSSION THREAD

**Discussion by slidebug** [ImageMagick/ImageMagick#3829] [Help] [answered]
Hey, 
I'm kinda new to Imagemagick and im  compress ".png" files with the Lanczos Filter.
Is the Lanczos Filter maybe a bad decision?

I have no clue about filters, so I'm not sure if the Lanczos Filter is a good decision :)

BTW: I'm using gopkg.in/gographics/imagick.v3/imagick.

Kindly regards

**Comment by snibgo** [comment]
I don't know how you compress PNG with the Lanczos filter. Please show the command you are using.

See https://imagemagick.org/script/defines.php for PNG options. Lanczos is not listed among them.

**Reply by slidebug** [reply]
`mw := imagick.NewMagickWand()`
`mw.ResizeImage(width, height, imagick.FILTER_LANCZOS)`

**Comment by snibgo** [comment]
Ah, you are using Lanczos for resizing. Under some circumstances, IM uses that filter by default.

What advantage do you get by forcing it to use Lanczos? I suspect there is no advantage.

**Reply by slidebug** [reply]
So actually the package that is provided for Golang needs this value.

I thought that the filter is doing some compression too, but I guess I'm wrong?

**Comment by snibgo** [comment, marked as answer]
The filter used has no effect on the number of pixels that ResizeImage makes.

The values of those pixels will influence the size in bytes of a compressed file. For example, PNG compresses better when many pixels have the same value. But there is no simple relationship between the filter and the final filesize.

**Reply by slidebug** [reply]
Thank you a lot!
I misunderstood the function.
